overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -69,20 +69,21 @@ func getServiceInfos(st *state.State, snapName string, serviceNames []string) ([
 
 var servicestateControl = servicestate.Control
 
-func queueCommand(context *hookstate.Context, ts *state.TaskSet) {
-	// queue command task after all existing tasks of the hook's change
-	st := context.State()
-	st.Lock()
-	defer st.Unlock()
+func queueCommand(context *hookstate.Context, ts *state.TaskSet) error {
+	context.Lock()
+	defer context.Unlock()
 
-	task, ok := context.Task()
+	hookTask, ok := context.Task()
 	if !ok {
-		panic("attempted to queue command with ephemeral context")
+		return fmt.Errorf("attempted to queue command with ephemeral context")
 	}
-	change := task.Change()
-	tasks := change.Tasks()
+
+	change := hookTask.Change()
+	tasks := change.LaneTasks(hookTask.Lanes()...)
 	ts.WaitAll(state.NewTaskSet(tasks...))
```

Comment:
I think we also want to have all the tasks in ts to join the same lanes as hooTask, unless those are just [0]

Reply:
do we need extra unit tests for this bit?